distribution/lib/Standard/Microsoft/0.0.0-dev/src/Internal/SQLServer_Type_Mapping.enso

```diff
@@ -0,0 +1,188 @@
+       the database backend.
+    infer_return_type : (SQL_Expression -> SQL_Type_Reference) -> Text -> Vector -> SQL_Expression -> SQL_Type_Reference
+    infer_return_type infer_from_database_callback op_name arguments expression =
+        _ = [op_name, arguments]
+        infer_from_database_callback expression
+
+    ## PRIVATE
+       We want to respect any overriding references, but references that rely on
+       computing the type by the database are resolved to Nothing to just rely
+       on the `ResultSet` metadata and decrease overhead.
+    prepare_type_overrides : Nothing | Vector SQL_Type_Reference -> Nothing | Vector (Nothing | SQL_Type)
+    prepare_type_overrides column_type_suggestions = case column_type_suggestions of
+        Nothing -> Nothing
+        _ : Vector -> column_type_suggestions.map .to_type_override
+
+    ## PRIVATE
+       Creates a `Column_Fetcher` used to fetch data from a result set and build
+       an in-memory column from it, based on the given column type.
+    make_column_fetcher : SQL_Type -> Column_Fetcher
+    make_column_fetcher self sql_type =
+        value_type = self.sql_type_to_value_type sql_type
+        Column_Fetcher_Module.default_fetcher_for_value_type value_type
+
+    ## PRIVATE
+    is_implicit_conversion (source_type : Value_Type) (target_type : Value_Type) -> Boolean =
+        # Currently, we do not have any implicit conversions.
+        _ = [source_type, target_type]
+        False
+
+    ## PRIVATE
+    should_warn_on_materialize (db_type : Value_Type) (in_memory_type : Value_Type) -> Boolean =
+        SQL_Type_Mapping.default_should_warn_on_materialize db_type in_memory_type
+
+## PRIVATE
+complex_types_map = Dictionary.from_vector <|
+    make_decimal sql_type =
+        Value_Type.Decimal sql_type.precision sql_type.scale
+    make_varchar sql_type =
+        effective_size = if sql_type.precision==max_length || (sql_type.precision==9 && sql_type.scale==9) then Nothing else sql_type.precision
+        Value_Type.Char size=effective_size variable_length=True
+    make_char sql_type =
+        Value_Type.Char size=sql_type.precision variable_length=False
+    make_binary variable sql_type =
+        Value_Type.Binary size=sql_type.precision variable_length=variable
+    handle_bit sql_type =
+        if sql_type.name == "BOOLEAN" then Value_Type.Boolean else
+            # We currently do not support bit types.
+            on_unknown_type sql_type
+    handle_timestamp sql_type = case sql_type.name of
+        "TIMESTAMPTZ" -> Value_Type.Date_Time with_timezone=True
+        "TIMESTAMP_TZ" -> Value_Type.Date_Time with_timezone=True
+        "TIMESTAMPLTZ" -> Value_Type.Date_Time with_timezone=False
+        "TIMESTAMP_LTZ" -> Value_Type.Date_Time with_timezone=False
+        "TIMESTAMPNTZ"   -> Value_Type.Date_Time with_timezone=False
+        "TIMESTAMP_NTZ"   -> Value_Type.Date_Time with_timezone=False
+        _             -> on_unknown_type sql_type
```

Comment:
After a brief look at T-SQL data types, this seems off. Isn't this a copy-paste from Snowflake?

I couldn't find these names used in T-SQL - it rather uses `DATETIMEOFFSET`/`DATETIME2` like you used above.

OK to address later.

Reply:
Yes left over from the snowflake starting point :)

Reply:
Removed